logger added

RoztX opened the pull request.

RoztX committed:
logger added

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Logger.cs:
```diff
@@ -0,0 +1,53 @@
+
+        public void Log(DateTime dateTime, LogType logType, string message)
+        {
+            var fullMessage = $"{dateTime} {logType} {message}";
```
Формат должен быть с ":"

RoztX replied:
Fixed

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Program.cs:
```diff
@@ -0,0 +1,6 @@
+
+    Starter starter = new Starter();
+    int logQuantity = 100;
+    starter.Start(logQuantity);
```
Цитирую задание:
"Создать в отдельном файле класс Starter с методом Run. Сигнатура метода ничего не принимает и не возвращает". Обрати внимание на имя твоего метода и его параметры на вход.

RoztX replied:
Fixed

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Logger.cs:
```diff
@@ -0,0 +1,53 @@
+            var nameFile = $"logs.txt";
+            for (int i = 0; i < logQuantity;i++)
+            {
+                File.AppendAllText(nameFile, _logs[i]);
```
Цитирую задание: "Запись в файл сделать через File.WriteAllText("log.txt", {полученный_лог});". Придерживайся задания.

RoztX replied:
Fixed

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Starter.cs:
```diff
@@ -0,0 +1,35 @@
+            {
+                Random random = new Random();
+                typeOfLog = random.Next(0, 3);
+                if (typeOfLog == 0)
```
Конструкция switch как раз для таких случаев

RoztX replied:
Fixed

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Actions.cs:
```diff
@@ -0,0 +1,27 @@
+        }
+        public Result Third()
+        {
+            Logger.Instance.Log(DateTime.Now, LogType.Error, $"Broke");
```
Не могу найти где в задании это указано. Про возврат стринги "Broke" не оговаривалось.

RoztX replied:
Fixed

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Actions.cs:
```diff
@@ -0,0 +1,27 @@
+        public Result Third()
+        {
+            Logger.Instance.Log(DateTime.Now, LogType.Error, $"Broke");
+            return new Result { status = false };
```
Цитирую задание: "Метод возвращает Result где свойство об ошибке = “I broke a logic”. а Status = false
". Необходимо выполнить все условия задания.

RoztX replied:
Fixed

RoztX committed:
1 добавил ":" в формат лога
2 изменил метод в Стартер
3 заменил AppendAllText на File.WriteAllText
4 заменил иф/эльсе на switch/case
5 отредактировал возвращаемое значение в методе Third

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Actions.cs:
```diff
@@ -0,0 +1,27 @@
+        }
+        public Result Third()
+        {
+            Logger.Instance.Log(DateTime.Now, LogType.Error, $"{nameof(Third)}");
```
Если внимательно перечитаешь задание, то убедишься, что в описании третьего метода класса Actions нет логирования в консоль

RoztX replied:
Fixed

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Logger.cs:
```diff
@@ -0,0 +1,54 @@
+
+        public void Log(DateTime dateTime, LogType logType, string message)
+        {
+            var fullMessage = $"{dateTime} : {logType} : {message}";
```
Маленькая ремарка - дроеточие без пробела слева и с пробелом справа FYI

RoztX replied:
Fixed

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Logger.cs:
```diff
@@ -0,0 +1,54 @@
+
+        public void SavelogToFile()
+        {
+            string container ="";
```
Почитай про класс StringBuilder на msdn. Твой пример работает, но стринг билдер создан как раз для таких целей.

RoztX replied:
Fixed

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Program.cs:
```diff
@@ -0,0 +1,5 @@
+
+    Starter starter = new Starter();
+    starter.Run();
+    Logger.Instance.SavelogToFile();
```
В задании в описании метода Run сказано "По окончанию цикла сформировать файл в котором будет текст всех записанных логов". Почему в таком случае логгер пишет в файл в классе Program?

RoztX replied:
Fixed

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Starter.cs:
```diff
@@ -0,0 +1,34 @@
+        public void Run()
+        {
+            Actions actions = new Actions();
+            var typeOfLog = 0;
```
Тебе в дальнейшем нужна эта переменная? (Если нет то можешь просто в свиче указать свой random.Next(3))

RoztX replied:
Fixed

RoztX committed:
- Убрал логирование из метода Third()
- Убрал пробел перед ":" в формате записи лога
- Добавил StringBuilder
- Перенес формирование файла в метод Run()
- Убрал лишнюю переменную из метода Run()

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Logger.cs:
```diff
@@ -0,0 +1,44 @@
+
+        static Logger()
+        {
+
```
stylecop такое не пропустит, чтобы оставались пусстые строки. Просто для твоей информации, так как в дальнейших домашках стайлкоп обязателен

RoztX replied:
Понял

DmytroZhuravlov14 commented on HomeWorke2.1/HomeWorke2.1/Starter.cs:
```diff
@@ -0,0 +1,35 @@
+            for (int i = 0; i < 100; i++)
+            {
+                Random random = new Random();
+                switch (random.Next(0, 3))
```
чуток можно укоротить как я писал ранее - random.Next(3), должно работать так же

RoztX replied:
Принял
